### PR TITLE
Added latency metrics for REST endpoint.

### PR DESCRIPTION
**Pull request by fulmicoton**
This PR also makes it possible to set specific buckets for histograms.

Cloes #4932

### CODE REVIEW

File: quickwit/quickwit-search/src/metrics.rs [hunk 1]
```diff
@@ -40,6 +42,7 @@ impl Default for SearchMetrics {
                 "Number of seconds required to run a leaf search over a single split. The timer \
                  starts after the semaphore is obtained.",
                 "search",
+                exponential_buckets(0.002, 2.0, 8).unwrap(),
```
Comment (trinity-1686a) [inline quickwit/quickwit-search/src/metrics.rs:45]: last bucket ends at half a second. while i understand smaller steps for leaf_search_split, in my experience having some split being longer than that isn't necessarily rare. imo either the start bucket or the number of bucket needs to be increased

File: quickwit/quickwit-serve/src/metrics.rs [hunk 2]
```diff
@@ -34,6 +35,14 @@ impl Default for RestMetrics {
                 &[],
                 ["method", "status_code"],
             ),
+            request_duration_seconds: new_histogram_vec(
+                "request_duration_seconds",
+                "Response time in millisecs",
```
Comment (trinity-1686a) [inline quickwit/quickwit-serve/src/metrics.rs:40]: we use `secs`, not seconds, elsewhere.
the comment gives the wrong unit too

**Review by trinity-1686a — APPROVED** [review]
lgtm, just a few small changes

File: quickwit/quickwit-serve/src/metrics.rs (hunk 2 above)
Comment (fulmicoton) [reply quickwit/quickwit-serve/src/metrics.rs:40]: good point. Yeah this is tricky. We used secs in all place but one (apart from here) and prometheus docs recommends seconds

Comment (fulmicoton) [reply quickwit/quickwit-serve/src/metrics.rs:40]: Ill switch to secs.

File: quickwit/quickwit-search/src/metrics.rs (hunk 1 above)
Comment (fulmicoton) [reply quickwit/quickwit-search/src/metrics.rs]: did both

**Pull request merged**